tests/unittests/GraphOptzTest.cpp

```diff
@@ -2707,37 +2733,20 @@ TEST_F(GraphOptz, sinkTransposeBelowChannelShuffleNodesAndEliminate) {
   K = F_->createTranspose("unnecessary_transpose_2", K, {0, 2, 3, 1});
   auto *save = F_->createSave("ret", K);
 
-  EXPECT_EQ(F_->getNodes().size(), 6);
+  EXPECT_EQ(F_->getNodes().size(), 4);
 
   // Optimize away the unnecessary transposes.
   optimize(F_, CompilationMode::Infer);
 
   // Ensure the two unnecessary transposes are gone.
-  ASSERT_EQ(F_->getNodes().size(), 4);
-
-  // Check that the channel shuffle nodes are still there.
-  auto *postShuffleRN = llvm::dyn_cast<ReshapeNode>(save->getInput().getNode());
-  ASSERT_NE(nullptr, postShuffleRN);
-  auto *shuffleTR =
-      llvm::dyn_cast<TransposeNode>(postShuffleRN->getInput().getNode());
-  ASSERT_NE(nullptr, shuffleTR);
-  auto *preShuffleRN =
-      llvm::dyn_cast<ReshapeNode>(shuffleTR->getInput().getNode());
-  ASSERT_NE(nullptr, preShuffleRN);
-
-  // Ensure last reshape has the same dimensions as the input.
-  EXPECT_EQ(postShuffleRN->getDims(), llvm::makeArrayRef(inputDims));
-
-  // Ensure the transpose in the middle of the reshapes shuffles the last
-  // dimension of the input.
-  const unsigned_t expectedShuffleTRShuffle[] = {0, 1, 2, 4, 3};
-  EXPECT_EQ(shuffleTR->getShuffle(),
-            llvm::makeArrayRef(expectedShuffleTRShuffle));
-
-  // Ensure the first reshape expands the last dimension of the input.
-  const size_t expectedPreShuffleRNDims[] = {3, 28, 28, 4, 34};
-  EXPECT_EQ(preShuffleRN->getDims(),
-            llvm::makeArrayRef(expectedPreShuffleRNDims));
+  ASSERT_EQ(F_->getNodes().size(), 2);
+
+  // Check that the channel shuffle node is still there.
+  auto *CSN = llvm::dyn_cast<ChannelShuffleNode>(save->getInput().getNode());
+  ASSERT_NE(nullptr, CSN);
+
+  // Ensure ChannelShuffle node has the same dimensions as the input.
+  EXPECT_EQ(CSN->getType(0)->dims(), llvm::makeArrayRef(inputDims));
```

Comment:
Same as in other test: `CS->getResult().dims()` here, and check that group and kernel are set as expected?